### PR TITLE
Add content to BLE.md

### PR DESCRIPTION
**Pull request by AnotherButler**
WIP: DO NOT MERGE
Add existing content.

### CODE REVIEW

File: docs/reference/api/connectivity/bluetooth/BLE.md [hunk 1]
```diff
@@ -4,6 +4,50 @@ Bluetooth low energy (BLE) is a low power wireless technology standard for build
...
+
+1. Interfaces under **'public/'** to express BLE constructs, such as GAP, GATT, services and characteristics.
+
+2. Code under **'common/'** encapsulates headers that need to be shared between the public interfaces and underlying bridge code.
+
+3. Classes under **'services/'** to offer reference implementations for many of the commonly used GATT profiles. The code under 'services/' isn't essential, but it’s a useful starting point for prototyping. We continue to implement the standard GATT profiles.
+
+#### The BLEDevice class and header
+
+The heart of Mbed's `BLE_API` is the `BLEDevice` class, accessible in the IDE using the header `BLEDevice.h`. This class allows us to create a BLE object that includes the basic attributes of a spec-compatible BLE device and can work with any BLE radio:
```
Comment (pan-) [inline docs/reference/api/connectivity/bluetooth/BLE.md]: I would rephrase the introduction into: _The entry point of Mbed's `BLE_API` is the BLE class_ accessible using the header `ble/BLE.h`. This class allows us to obtain a BLE object that includes the basic attributes of a spec-compatible BLE device and can work with any BLE radio:_

File: docs/reference/api/connectivity/bluetooth/BLE.md [hunk 2]
```diff
@@ -4,6 +4,50 @@ Bluetooth low energy (BLE) is a low power wireless technology standard for build
...
+
+```c
+#include "BLEDevice.h"
```
Comment (pan-) [inline docs/reference/api/connectivity/bluetooth/BLE.md]: `#include "ble/BLE.h"`

File: docs/reference/api/connectivity/bluetooth/BLE.md [hunk 3]
```diff
@@ -4,6 +4,50 @@ Bluetooth low energy (BLE) is a low power wireless technology standard for build
...
+
+BLEDevice mydevicename;
```
Comment (pan-) [inline docs/reference/api/connectivity/bluetooth/BLE.md]: `BLE& mydevicename = BLE::Instance();`

File: docs/reference/api/connectivity/bluetooth/BLE.md [hunk 4]
```diff
@@ -4,6 +4,50 @@ Bluetooth low energy (BLE) is a low power wireless technology standard for build
...
+```
+
+The class's member functions can be divided by purpose:
+
+1. Basic BLE operations, such as initializing the controller.
+
+1. GAP related methods: radio transmission power levels, advertisements and parameters affecting connection establishment.
```
Comment (pan-) [inline docs/reference/api/connectivity/bluetooth/BLE.md]: We can remove the GAP and GATT related bit and transform it into: 

`1. Accessor to Bluetooth Modules that manage GAP, GATT or the security.`

File: docs/reference/api/connectivity/bluetooth/BLE.md [hunk 5]
```diff
@@ -4,6 +4,50 @@ Bluetooth low energy (BLE) is a low power wireless technology standard for build
...
+
+1. GATT related methods: to set up and control the GATT database describing services together with the characteristics and attributes composing them.
+
+1. Event-driven programming controls, such as methods to set up various callbacks to invoke in response to system events.
```
Comment (pan-) [inline docs/reference/api/connectivity/bluetooth/BLE.md]: This can be removed.

**Comment by AnotherButler** [comment]
@pan- Do my changes address your comments?
